### ISSUE TITLE
Error running prepare.sh

### ISSUE THREAD

**Issue by dachafra** [anifledramos/tfm-rdf#5]
```
MacBook-Pro-de-David:tfm-rdf dchaves$ ./prepare.sh
Clonando en 'tfm-rdf'...
remote: Enumerating objects: 2363, done.
remote: Counting objects: 100% (2363/2363), done.
remote: Compressing objects: 100% (1317/1317), done.
remote: Total 2363 (delta 1124), reused 1980 (delta 744), pack-reused 0
Recibiendo objetos: 100% (2363/2363), 45.96 MiB | 2.77 MiB/s, listo.
Resolviendo deltas: 100% (1124/1124), listo.
rm -f obj/mymalloc.o obj/merge.o obj/basic.o obj/bitrankw32int.o obj/directcodesI.o obj/directcodes.o obj/kTreeAux.o obj/kTreeCreation.o obj/kTree.o  obj/interfazSO.o obj/kTreeJoin.o obj/kTreeJoinInt.o obj/hash.o obj/MemoryManager.o obj/joinsFF.o obj/joinsFV.o obj/joinsVV.o obj/simplePatternsF.o obj/simplePatterns.o obj/k2triples.o obj/commons-indices.o obj/k2triples-builder.o i build_tree use_tree
cd bin; rm -f *.a build_tree use_tree
gcc -Wall -g -fPIC -shared  -c src/mymalloc.c -o obj/mymalloc.o
clang: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
gcc -Wall -g -fPIC -shared  -c src/merge.c -o obj/merge.o
clang: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
In file included from src/merge.c:1:
src/merge.h:12:3: error: unknown type name 'uint'; did you mean 'int'?
{ uint* sources;
  ^~~~
  int
src/merge.h:13:3: error: unknown type name 'uint'; did you mean 'int'?
  uint value;
  ^~~~
  int
src/merge.c:44:19: warning: assigning to 'int *' from 'unsigned int *' converts between pointers to integer types with different sign [-Wpointer-sign]
                                v1[i].sources = (unsigned int*) realloc(v1[i].sources, (n_sources + 1) * sizeof(unsigned int));
                                              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 2 errors generated.
make: *** [obj/merge.o] Error 1
cp: /usr/lib/libjnitriples.so: Operation not permitted
g++ -std=c++11 -shared -fPIC -O3 -I/Library/Java/JavaVirtualMachines/jdk1.8.0_281.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk1.8.0_281.jdk/Contents/Home/include/linux -I libCSD/libcds/includes src/org_rdfhdt_hdt_dictionary_impl_section_JNIDictionarySection.cpp ./libCSD/dictionary.a -o lib/libjnidictionary.so
In file included from src/org_rdfhdt_hdt_dictionary_impl_section_JNIDictionarySection.cpp:1:
In file included from src/org_rdfhdt_hdt_dictionary_impl_section_JNIDictionarySection.h:2:
/Library/Java/JavaVirtualMachines/jdk1.8.0_281.jdk/Contents/Home/include/jni.h:45:10: fatal error: 'jni_md.h' file not found
#include "jni_md.h"
         ^~~~~~~~~~
1 error generated.
make: *** [jni] Error 1
cp: /usr/lib/libjnidictionary.so: Operation not permitted
```

```
[INFO] ----------------------< org.rdfhdt:hdt-java-cli >-----------------------
[INFO] Building HDT Java Command line Tools 2.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ hdt-java-cli ---
Converting ../../nt/test.nt to ../../nt/test.nt as null
Max memory (bytes): 11453595648
Free memory (bytes): 436244528
Free memory (bytes): 436244528
Free memory (bytes): 434928464
Free memory (bytes): 434928464
Free memory (bytes): 434928464
Free memory dictionary init (bytes): 434928464
[WARNING]
java.lang.UnsatisfiedLinkError: no jnidictionary in java.library.path
    at java.lang.ClassLoader.loadLibrary (ClassLoader.java:1860)
    at java.lang.Runtime.loadLibrary0 (Runtime.java:871)
    at java.lang.System.loadLibrary (System.java:1122)
    at org.rdfhdt.hdt.dictionary.impl.section.JNIDictionarySection.<clinit> (JNIDictionarySection.java:62)
    at org.rdfhdt.hdt.dictionary.impl.JNIDictionary.<init> (JNIDictionary.java:36)
    at org.rdfhdt.hdt.dictionary.DictionaryFactory.createDictionary (DictionaryFactory.java:82)
    at org.rdfhdt.hdt.hdt.impl.HDTImpl.createComponents (HDTImpl.java:94)
    at org.rdfhdt.hdt.hdt.impl.HDTImpl.<init> (HDTImpl.java:138)
    at org.rdfhdt.hdt.hdt.HDTManagerImpl.doGenerateHDT (HDTManagerImpl.java:108)
    at org.rdfhdt.hdt.hdt.HDTManager.generateHDT (HDTManager.java:129)
    at org.rdfhdt.hdt.tools.RDF2HDT.execute (RDF2HDT.java:122)
    at org.rdfhdt.hdt.tools.RDF2HDT.main (RDF2HDT.java:238)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:748)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.816 s
[INFO] Finished at: 2021-06-15T08:58:49+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:java (default-cli) on project hdt-java-cli: An exception occured while executing the Java class. no jnidictionary in java.library.path -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
+ exit 1
```

Finally it seems that it works, but some errors or warning appeared

**Comment by anifledramos** [comment]
I think these issues are related to the OS. The development of this project was made specifically for Linux, following its data structures and so. 

The issue with JNITriples is due to the implementation of K2Triples in C, which was made by a colleague before me, working with Linux as well. 

The issue with `hdt-java-cli` is related to the library types of each OS. In Linux we work with .so, in Windows with .dll and in MacOS with .dylib. I don't really know if it is possible to translate everything to work on Mac since we would have to touch the implementation of the compression algorithms and all. In addition, I have no MacOS to replicate and test this issue.

I discussed possible solutions with my tutor and we concluded that it should run on Linux, no adaptation for other OS will be provided (at least for now).
